code/_onclick/hud/alert.dm

```diff
@@ -51,13 +51,17 @@
 	thealert.owner = src
 
 	if(new_master)
-		var/old_layer = new_master.layer
-		var/old_plane = new_master.plane
-		new_master.layer = FLOAT_LAYER
-		new_master.plane = FLOAT_PLANE
-		thealert.add_overlay(new_master)
-		new_master.layer = old_layer
-		new_master.plane = old_plane
+		// This should be made into a generic helper proc later tbh
+		var/mutable_appearance/master_appearance = new
+		master_appearance.appearance = new_master.appearance
+		master_appearance.appearance_flags = KEEP_TOGETHER
+		master_appearance.layer = FLOAT_LAYER
+		master_appearance.plane = FLOAT_PLANE
+		master_appearance.dir = SOUTH
+		master_appearance.pixel_x = 0
+		master_appearance.pixel_y = 0
+		master_appearance.pixel_z = 0
```

Comment:
thought: what about transform, and what about mobs that are larger then 32x32 that get shifted over to make sense visually (megafauna) 
is this actually something we want

Reply:
hmm you're right... i should be setting it to base_pixel_x/y, not 0